process/peer/process.go

```diff
@@ -122,13 +123,98 @@ func (p *validatorStatistics) IsNodeValid(node *sharding.InitialNode) bool {
 	return true
 }
 
+func (p *validatorStatistics) processPeerChanges(header data.HeaderHandler) error {
+	if p.shardCoordinator.SelfId() == sharding.MetachainShardId {
+		return nil
+	}
+
+	metaBlock, ok := header.(*block.MetaBlock)
+	if !ok {
+		return nil
+	}
+
+	for _, peerChange := range metaBlock.PeerInfo {
+		err := p.updatePeerState(peerChange)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *validatorStatistics) updatePeerState(
+	peerChange block.PeerData,
+) error {
+	adrSrc, err := p.adrConv.CreateAddressFromPublicKeyBytes(peerChange.Address)
+	if err != nil {
+		return err
+	}
+
+	if peerChange.Action == block.PeerDeregistration {
+		return p.peerAdapter.RemoveAccount(adrSrc)
+	}
+
+	accHandler, err := p.peerAdapter.GetAccountWithJournal(adrSrc)
+	if err != nil {
+		return err
+	}
+
+	account, ok := accHandler.(*state.PeerAccount)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	if !bytes.Equal(peerChange.PublicKey, account.BLSPublicKey) {
+		err := account.SetBLSPublicKeyWithJournal(peerChange.PublicKey)
+		if err != nil {
+			return err
+		}
+	}
+
+	zero := big.NewInt(0)
+	if peerChange.ValueChange.Cmp(zero) != 0 {
+		actualValue := zero.Add(account.Stake, peerChange.ValueChange)
+		err := account.SetStakeWithJournal(actualValue)
+		if err != nil {
+			return err
+		}
+	}
+
+	if peerChange.Action == block.PeerRegistrantion && peerChange.TimeStamp != account.Nonce {
```

Comment:
We use the timestamp as peer account's nonce? So counter intuitive.

Reply:
yeah, timeStamp is equal with nonce, maybe it should be round.

Reply:
It remains counter intuitive. Why didn't we have a specialized field called Timestamp or Round (just like Balance in the plain Account)?

Reply:
Can you put at least a TODO there to refactor this